clients/src/main/java/org/apache/kafka/common/network/KafkaChannel.java

```diff
@@ -72,23 +72,16 @@ public KafkaPrincipal principal() {
      * Does handshake of transportLayer and authentication using configured authenticator
      */
     public void prepare() throws IOException {
-        if (!transportLayer.ready())
-            transportLayer.handshake();
-        if (transportLayer.ready() && !authenticator.complete()) {
-            try {
+        try {
+            if (!transportLayer.ready())
+                transportLayer.handshake();
```

Comment:
Can we please add some documentation to `Authenticator.authenticate` and `TransportLayer.handshake` explaining both can actually `authenticate` depending on the configuration. Worth updating the documentation of `prepare` to make that clear too.

Reply:
@ijuma Thank you for the review. Added comment.